### PR TITLE
Tests utils

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
- added tests for methods in flow/utils
- bug fix to the assignment of traffic lights in `get_flow_params` method in utils/rllib.py
- modified the `types` attribute of the `Vehicles` class for it to be more easily unittestable

### CODE REVIEW

File: tests/fast_tests/test_util.py [hunk 1]
```diff
@@ -68,5 +76,237 @@ class Foo(object):
...
+        # call make_create_env
+        create_env, env_name = make_create_env(params=flow_params, version=v)
+
+        # check that the name is correct
+        self.assertEqual(env_name, '{}-v{}'.format(flow_params["env_name"], v))
+
+        # create the gym environment
+        env = create_env()
+
+        # Note that we expect the port number is sumo_params to change, and
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_util.py]: port number "in"

File: tests/fast_tests/test_util.py [hunk 2]
```diff
@@ -68,5 +76,237 @@ class Foo(object):
...
+
+        # check that each of the parameter match
+        self.assertEqual(env.env.env_params.__dict__,
+                         flow_params["env"].__dict__)
+        self.assertEqual(env.env.sumo_params.__dict__,
+                         flow_params["sumo"].__dict__)
+        self.assertEqual(env.env.traffic_lights.__dict__,
+                         flow_params["tls"].__dict__)
+        self.assertEqual(env.env.scenario.net_params.__dict__,
+                         flow_params["net"].__dict__)
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_util.py]: Empirically I've observed errors with saving of the netfile when there are inflows. Can we reconfigure this test to include inflows?

File: tests/fast_tests/test_util.py [hunk 3]
```diff
@@ -68,5 +76,237 @@ class Foo(object):
...
+            env_name="AccelEnv",
+            scenario="Figure8Scenario",
+            generator="Figure8Generator",
+            sumo=SumoParams(
+                sim_step=0.1,
+                sumo_binary="sumo",
+            ),
+            env=EnvParams(
+                horizon=1500,
+                additional_params={
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_util.py]: Same here as above.

File: tests/fast_tests/test_vehicles.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def test_controlled_id_params(self):
                      acceleration_controller=(IDMController, {}),
                      speed_mode='no_collide',
                      lane_change_mode="no_lat_collide")
-        self.assertEqual(vehicles.types[0][1]["minGap"], 0)
+        self.assertEqual(vehicles.types[0]["type_params"]["minGap"], 0)
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_vehicles.py]: Nice, no more floating magic numbers.

File: tests/fast_tests/test_util.py (hunk 2 above)
Comment (eugenevinitsky) [reply tests/fast_tests/test_util.py]: We want to be checking against the most complex environment possible, and that includes inflows.

Comment (AboudyKreidieh) [reply tests/fast_tests/test_util.py]: done

File: tests/fast_tests/test_util.py (hunk 1 above)
Comment (AboudyKreidieh) [reply tests/fast_tests/test_util.py]: fixed

**Pull request merged**